src/Microsoft.Diagnostics.Monitoring.WebApi/EgressValidationAttribute.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Extensions.Logging;
+using System;
+
+namespace Microsoft.Diagnostics.Monitoring.WebApi
+{
+    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Class)]
+    public sealed class EgressValidationAttribute : ActionFilterAttribute, IFilterFactory
+    {
+        public bool IsReusable => true;
+        IFilterMetadata IFilterFactory.CreateInstance(IServiceProvider serviceProvider)
+        {
+            var egressOutputOptions = serviceProvider.GetRequiredService<IEgressOutputOptions>();
+
+            EgressValidationActionFilter actionFilter = new EgressValidationActionFilter(egressOutputOptions);
+
+            return actionFilter;
+        }
+    }
+
+    public class EgressValidationUnhandledExceptionFilter : ActionFilterAttribute, IExceptionFilter
+    {
+        public void OnException(ExceptionContext context)
+        {
+            BadRequestObjectResult badRequestResult = new BadRequestObjectResult(context.Exception.Message);
+            badRequestResult.Value = ExceptionExtensions.ToProblemDetails(context.Exception, StatusCodes.Status400BadRequest);
+
+            context.Result = badRequestResult;
+        }
+    }
+
+    public class EgressValidationActionFilter : IActionFilter
+    {
+        private IEgressOutputOptions _egressOutputOptions;
+        private const string EgressQuery = "egressprovider";
+
+        public EgressValidationActionFilter(IEgressOutputOptions egressOutputOptions)
+        {
+            _egressOutputOptions = egressOutputOptions;
+        }
+
+        public void OnActionExecuted(ActionExecutedContext context)
+        {
+            // Not sure we need to do anything here...
+        }
+
+        public void OnActionExecuting(ActionExecutingContext context)
+        {
+            StringValues value;
+
+            bool egressProviderGiven = context.HttpContext.Request.Query.TryGetValue(EgressQuery, out value);
+
+            if (!egressProviderGiven || value.ToString().Equals(""))
+            {
+                if (_egressOutputOptions.EgressMode == EgressMode.HttpDisabled)
+                {
+                    ILogger<EgressValidationActionFilter> logger = context.HttpContext.RequestServices
```

Comment:
Actually, I think all of the logging should move to `EgressValidationUnhandledExceptionFilter`. This will keep the same pattern we have elsewhere (logging is done at the site of converting an exception to problem details), and you should be able to use DI directly rather than getting the logger factory and creating the logger yourself (update the `EgressValidationUnhandledExceptionFilter` constructor to have the ILogger parameter).